model/post.go

```diff
@@ -18,56 +18,58 @@ import (
-	POST_PROPS_OVERRIDE_ICON_EMOJI = "override_icon_emoji"
-
-	POST_PROPS_MENTION_HIGHLIGHT_DISABLED = "mentionHighlightDisabled"
-	POST_PROPS_GROUP_HIGHLIGHT_DISABLED   = "disable_group_highlight"
-	POST_SYSTEM_WARN_METRIC_STATUS        = "warn_metric_status"
+	PostSystemMessagePrefix        = "system_"
+	PostTypeDefault                = ""
+	PostTypeSlackAttachment        = "slack_attachment"
+	PostTypeSystemGeneric          = "system_generic"
+	PostTypeJoinLeave              = "system_join_leave" // Deprecated, use PostJoinChannel or PostLeaveChannel instead
+	PostTypeJoinChannel            = "system_join_channel"
+	PostTypeGuestJoinChannel       = "system_guest_join_channel"
+	PostTypeLeaveChannel           = "system_leave_channel"
+	PostTypeJoinTeam               = "system_join_team"
+	PostTypeLeaveTeam              = "system_leave_team"
+	PostTypeAutoResponder          = "system_auto_responder"
+	PostTypeAddRemove              = "system_add_remove" // Deprecated, use PostAddToChannel or PostRemoveFromChannel instead
+	PostTypeAddToChannel           = "system_add_to_channel"
+	PostTypeAddGuestToChannel      = "system_add_guest_to_chan"
+	PostTypeRemoveFromChannel      = "system_remove_from_channel"
+	PostTypeMoveChannel            = "system_move_channel"
+	PostTypeAddToTeam              = "system_add_to_team"
+	PostTypeRemoveFromTeam         = "system_remove_from_team"
+	PostTypeHeaderChange           = "system_header_change"
+	PostTypeDisplaynameChange      = "system_displayname_change"
+	PostTypeConvertChannel         = "system_convert_channel"
+	PostTypePurposeChange          = "system_purpose_change"
+	PostTypeChannelDeleted         = "system_channel_deleted"
+	PostTypeChannelRestored        = "system_channel_restored"
+	PostTypeEphemeral              = "system_ephemeral"
+	PostTypeChangeChannelPrivacy   = "system_change_chan_privacy"
+	PostTypeAddBotTeamsChannels    = "add_bot_teams_channels"
+	PostTypeSystemWarnMetricStatus = "warn_metric_status"
+	PostTypeMe                     = "me"
+	PostCustomTypePrefix           = "custom_"
+
+	PostFileidsMaxRunes   = 300
+	PostFilenamesMaxRunes = 4000
+	PostHashtagsMaxRunes  = 1000
+	PostMessageMaxRunesV1 = 4000
+	PostMessageMaxBytesV2 = 65535                     // Maximum size of a TEXT column in MySQL
+	PostMessageMaxRunesV2 = PostMessageMaxBytesV2 / 4 // Assume a worst-case representation
+	PostPropsMaxRunes     = 8000
+	PostPropsMaxUserRunes = PostPropsMaxRunes - 400 // Leave some room for system / pre-save modifications
+
+	PropsAddChannelMember = "add_channel_member"
+
+	PostPropsAddedUserId       = "addedUserId"
+	PostPropsDeleteBy          = "deleteBy"
+	PostPropsOverrideIconUrl   = "override_icon_url"
+	PostPropsOverrideIconEmoji = "override_icon_emoji"
+
+	PostPropsMentionHighlightDisabled = "mentionHighlightDisabled"
+	PostPropsGroupHighlightDisabled   = "disable_group_highlight"
 )
 
-var AT_MENTION_PATTEN = regexp.MustCompile(`\B@`)
+var AtMentionPatten = regexp.MustCompile(`\B@`)
```

Comment:
Finally, we can correct the spelling mistake here :) Let's change it to `AtMentionPattern`.

Reply:
Fixed in c9f571566